src/cgame/default/cg_hud.c

```diff
@@ -1175,6 +1233,231 @@ static void Cg_DrawTargetName(const player_state_t *ps) {
 /**
  * @brief
  */
+void Cg_ParseNotification(void) {
+	bg_notification_item_t *item = g_malloc0(sizeof(bg_notification_item_t));
+
+	item->type = cgi.ReadByte();
+
+	switch(item->type) {
+		case NOTIFICATION_TYPE_OBITUARY:
+			item->mod = cgi.ReadByte();
+			item->client_id_1 = cgi.ReadByte();
+			item->client_id_2 = cgi.ReadByte();
+
+			item->when = 6000;
+
+			if (!cg_draw_notifications_disable_print->value) {
+				cgi.Print("^7%s ^1%s ^7%s\n", cgi.client->client_info[item->client_id_1].name,
+					Bg_GetModString(item->mod, item->mod & MOD_FRIENDLY_FIRE),
+					cgi.client->client_info[item->client_id_2].name);
+			}
+
+			break;
+		case NOTIFICATION_TYPE_OBITUARY_SELF:
+			item->mod = cgi.ReadByte();
+			item->client_id_1 = cgi.ReadByte();
+
+			item->when = 6000;
+
+			if (!cg_draw_notifications_disable_print->value) {
+				cgi.Print("^1%s ^7%s\n", Bg_GetModString(item->mod, false),
+					cgi.client->client_info[item->client_id_1].name);
+			}
+
+			break;
+		case NOTIFICATION_TYPE_OBITUARY_PIC: {
+			const char *p = cgi.ReadString();
+			strcpy(item->pic, p);
```

Comment:
"pic" should be cgi.LoadImage'd. I'm not certain, but if the obituary pictures are entirely constant (that is, the pics you send are *always* consistent between game and cgame for obituaries), send the index to the array and not the pic itself. Then we can reference a cgame-specific array of the images that are precached in this file.